File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/sagemaker/src/main/java/com/example/sage/ListModels.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/


package com.example.sage;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.sagemaker.SageMakerClient;
import software.amazon.awssdk.services.sagemaker.model.ListModelsRequest;
import software.amazon.awssdk.services.sagemaker.model.ListModelsResponse;
import software.amazon.awssdk.services.sagemaker.model.ModelSummary;
import software.amazon.awssdk.services.sagemaker.model.SageMakerException;
import java.util.List;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */

public class ListModels {

    public static void main(String[] args) {

            Region region = Region.US_WEST_2;
            SageMakerClient sageMakerClient = SageMakerClient.builder()
                    .region(region)
                    .build();

            listAllModels(sageMakerClient);
            sageMakerClient.close();
        }

       public static void listAllModels(SageMakerClient sageMakerClient) {
            try {

                ListModelsRequest modelsRequest = ListModelsRequest.builder()
                        .maxResults(15)
                        .build();

                ListModelsResponse modelResponse = sageMakerClient.listModels(modelsRequest);
                List<ModelSummary> items = modelResponse.models();

                for (ModelSummary item : items) {
                    System.out.println("Model name is: " + item.modelName());
                }

            } catch (SageMakerException e) {
                System.err.println(e.awsErrorDetails().errorMessage());
                System.exit(1);
            }
        }
    }

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/sagemaker/src/main/kotlin/com/kotlin/sage/ListModels.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.sage

import aws.sdk.kotlin.services.sagemaker.SageMakerClient
import aws.sdk.kotlin.services.sagemaker.model.ListModelsRequest

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */
    suspend fun main() {

     listAllModels()
    }

    suspend fun listAllModels() {

         val request = ListModelsRequest {
                maxResults = 15
         }
        SageMakerClient { region = "us-west-2" }.use { sageMakerClient ->
           val response = sageMakerClient.listModels(request)
           response.models?.forEach { item ->
              println("Model name is ${item.modelName}")
           }
        }
   }

